internal/transformers/namedevaluation.go

```diff
@@ -0,0 +1,556 @@
+package transformers
+
+import (
+	"slices"
+
+	"github.com/microsoft/typescript-go/internal/ast"
+	"github.com/microsoft/typescript-go/internal/printer"
+)
+
+/**
+ * Gets whether a node is a `static {}` block containing only a single call to the `__setFunctionName` helper where that
+ * call's second argument is the value stored in the `assignedName` property of the block's `EmitNode`.
+ * @internal
+ */
+func isClassNamedEvaluationHelperBlock(emitContext *printer.EmitContext, node *ast.Node) bool {
+	if !ast.IsClassStaticBlockDeclaration(node) || len(node.AsClassStaticBlockDeclaration().Body.AsBlock().Statements.Nodes) != 1 {
+		return false
+	}
+
+	statement := node.AsClassStaticBlockDeclaration().Body.AsBlock().Statements.Nodes[0]
+	if ast.IsExpressionStatement(statement) {
+		expression := statement.AsExpressionStatement().Expression
+		if emitContext.IsCallToHelper(expression, "___setFunctionName") {
```

Comment:
Three underscores here; is this a typo?

Reply:
Huh, the original code has this too. Is this due to the old compiler's different symbol name escaping mechanism? The func generally seems like it should be just `__setFunctionName`, so...